### PR TITLE
[type/test][type/documentation] Add hosted-auth test coverage and documentation

### LINKED ISSUE #114: [Test] Add authentication coverage for GitHub App-first hosted auth
## Description

Add automated coverage for the GitHub App-first hosted-authentication tranche so the revised public-auth model can evolve safely.

## User Story
**As a** solo developer,
**I want** automated tests around GitHub App user sign-in, installation and token lifecycle handling, admission control, and the fallback boundary,
**So that** the hosted authentication path is safer to implement and regressions are caught early.

## Acceptance Criteria
- [ ] Test coverage expectations are defined for GitHub App user sign-in and per-request user context behaviour.
- [ ] Test coverage expectations are defined for installation discovery, token issuance, refresh, expiry, and failure handling.
- [ ] Admission-control and allow-list edge cases are identified explicitly.
- [ ] The plan identifies where mocks, integration seams, or environment-dependent tests are appropriate.
- [ ] Documentation-sensitive regressions are called out where the fallback path remains available.

## Related Feature / Milestone
Feature #103 / v1.0.0 — Production Ready

## Additional Context
This issue supersedes the earlier GitHub App plus OAuth coverage plan.

### LINKED ISSUE #119: [Docs] Update documentation for GitHub App-first hosted auth
## Description

Update the documentation set for the GitHub App-first hosted-authentication architecture, hosted admission control, and the retained PAT-only local trusted mode.

## Motivation / Context

ADR-0015 replaces the earlier implied hybrid GitHub App plus separate OAuth App direction. The implementation work needs matching documentation so that setup, deployment, and operator guidance all describe the same end state.

## Definition of Done
- [ ] Planning artefacts and ADR references stay consistent with the GitHub App-first direction.
- [ ] Getting-started and infrastructure guidance identify hosted prerequisites, allow-list administration expectations, and PAT-only local trusted mode clearly.
- [ ] User-facing documentation reflects the hosted sign-in model at the right level of detail.
- [ ] Any deprecated separate OAuth App guidance is removed or marked as fallback-only.

## Area

area/docs

### PR DESCRIPTION
**Pull request by markheydon**
## Summary

Closes #114
Closes #119

Delivers hosted-auth test coverage (issue \#114) and the new hosted authentication operator/user guide (issue \#119) together on a single branch.

## Type of Change

- [x] ✨ New feature
- [ ] 🐛 Bug fix
- [x] 📝 Documentation update
- [ ] 🔧 Chore / maintenance
- [ ] 🏗️ Infrastructure change

### Issue \#114 — Hosted Auth Test Coverage
- Added 4 new tests to `HostedGitHubAuthGatewayTests` covering:
  - Token payload containing an error field
  - User API response missing the login field
  - No matching owner installation (falls back to first installation)
  - Organisation response containing duplicate logins (returns distinct set)
- Added 2 new tests to `HostedUserCurrentUserContextTests` covering:
  - Missing expiry claim type in configuration
  - Missing owner-login claim type in configuration
- Added 2 new tests to `AllowListHostedAdmissionEvaluatorTests` covering:
  - Organisation claim value using mixed separators
  - Allow-list values using different casing (case-insensitive matching)
- Added Test Coverage Expectations section to `plan/HOSTED_AUTH_SESSION_AND_TOKEN_FLOW.md`

### Issue \#119 — Hosted Auth Documentation
- Created `docs/user-guide/hosted-authentication.md` (new operator/user guide)
- Updated `docs/getting-started.md` with hosted sign-in prerequisites, PAT-only local trusted mode, and OAuth App fallback sections
- Updated `infra/README.md` with hosted sign-in and admission control section; restored previously missing baseline assumptions content
- Updated `docs/index.md` and `docs/user-guide/index.md` with hosted authentication links
- Updated `plan/BACKLOG.md` and `plan/IMPLEMENTATION_PLAN.md` to mark both issues complete

## Quality Gates

- [x] Build passes with 0 errors, 0 warnings (`dotnet build`)
- [x] All 175 tests pass (`dotnet test`)
- [x] No compile-time errors in any modified test file
- [x] UK English verified throughout (no US spellings)
- [x] Layered architecture rules followed
- [x] xUnit + Moq used; FluentAssertions not introduced
- [x] All public members have XML doc comments
- [x] User-facing doc created (`docs/user-guide/hosted-authentication.md`)
- [x] `docs/index.md` and `docs/user-guide/index.md` updated
- [x] `plan/BACKLOG.md` and `plan/IMPLEMENTATION_PLAN.md` synced
- [x] No secrets or credentials in any committed file

## Testing

- 8 new test methods across 3 test files
- Test naming follows `MethodUnderTest_Scenario_ExpectedOutcome` convention
- Total test suite: 175 tests (58 Application + 61 Infrastructure + 50 App + 6 Domain)

## Documentation

- `docs/user-guide/hosted-authentication.md` — new hosted auth operator/user guide
- `docs/getting-started.md` — hosted sign-in prerequisites, PAT-only, OAuth fallback
- `infra/README.md` — hosted sign-in and admission control operator notes

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -30,7 +30,25 @@ SoloDevBoard supports two authentication modes:
 
 - **GitHub App:** Recommended for production deployments. Provides fine-grained permissions and does not expire. See the [GitHub Apps documentation](https://docs.github.com/en/apps) for setup instructions.
 
----
+
+#### Hosted Sign-In Prerequisites (Production)
+
+For production deployments, SoloDevBoard uses a GitHub App-first hosted authentication model:
+
+- Operators must configure the GitHub App and enable hosted sign-in in application settings (`HostedSignInEnabled=true`).
```
[the remark this answers is not in the document]
Comment (markheydon) [reply docs/getting-started.md]: Updated in this branch. The hosted sign-in prerequisite now references the correctly scoped environment variable key: `GitHubAuth__HostedSignInEnabled=true`.

File: infra/README.md [hunk 2]
```diff
@@ -23,6 +23,16 @@ The templates in this directory are configured for a public-release-ready baseli
...
 - Deployment authentication from GitHub Actions uses OIDC and repository environment protections.
 
+## Hosted Sign-In and Admission Control
+
+For production deployments using GitHub App-first hosted authentication:
+
+- Hosted sign-in should be enabled with `GitHubAuth__HostedSignInEnabled=true`.
+- Operators should configure `HostedAdmissionControl__Enabled=true` and maintain explicit user and organisation allow-lists.
+- Hosted access is deny-by-default when admission control is enabled; only configured allow-list identities are admitted.
+- PAT-only local trusted mode remains available for local development and trusted self-hosted use.
```
[the remark this answers is not in the document]
Comment (markheydon) [reply infra/README.md]: Checked this section against the documentation rule. The sentence-style bullets in this block now all end with full stops in the current branch state.
